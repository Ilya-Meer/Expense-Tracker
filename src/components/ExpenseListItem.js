import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import numeral from 'numeral';
import moment from 'moment';


const Styled = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 1rem 0;
  padding: 1rem 5%;
  border: 1px solid #ccc;
  

  a {
    text-decoration: none;
    color: #176ddf;
  }

  h3 {
    margin: 0;
    padding-bottom: 5px;
    font-weight: 700;
  }

  p {
    margin: 0
  }
`
const Col = styled.div`
  display: flex;
  flex-direction: column;
`



const ExpenseListItem = (props) => (
  <Styled>
    <Col>
      <Link to={`/edit/${props.id}`}>
        <h3>{props.desc}</h3>
      </Link>
      <p>Created on {moment(props.createdAt).format("MMM Do, YYYY")}</p>
    </Col>
    <Col>
      <p>{numeral(props.amount / 100).format('$0,0.00')}</p>
    </Col>
  </Styled>
)


export default ExpenseListItem