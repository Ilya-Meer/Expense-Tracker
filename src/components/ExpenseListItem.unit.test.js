import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from './ExpenseListItem';
import moment from 'moment';


describe('Expense List Item', () => {
  const expense = {
    id: '3',
    desc: 'Rent', 
    note: 'The rent was not cheap either', 
    amount: '2000000', 
    createdAt: moment(0)
  };

  it('should render expense list item', () => {
    const wrapper = shallow(<ExpenseListItem props={expense}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
