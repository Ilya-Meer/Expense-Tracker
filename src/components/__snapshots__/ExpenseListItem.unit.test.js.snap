// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List Item should render expense list item 1`] = `
<styled.div>
  <styled.div>
    <Link
      replace={false}
      to="/edit/undefined"
    >
      <h3 />
    </Link>
    <p>
      Created on 
      Dec 31st, 1969
    </p>
  </styled.div>
  <styled.div>
    <p>
      $0.00
    </p>
  </styled.div>
</styled.div>
`;
